tests/test_ancillaries.py

```diff
@@ -12,11 +12,20 @@
 def fake_af_cleanup_fixture():
     """Fixture used to setup and cleanup some fake ancillary features"""
     # code run before the test
+    # set the method calls of our shared method to zero (just in case)
+    shared_fake_af_method.calls = 0
+    # setup fake ancillary features
     af1, af2 = setup_fake_af()
     # variables yielded (input) to the test, then the test is run
     yield af1, af2
     # code run after the test
+    # set the method calls of our shared method to zero
     shared_fake_af_method.calls = 0
+    # remove our fake ancillary feature instances from the registry
```

Comment:
Thought this might be the case..., but this makes me realise that the `yield` fixture from pytest isn't exactly a context manager, right? @paulmueller

Reply:
Are you worried about cleanup after failed tests? I think pytest also executes the code after yield if the test raised an assertion.

Having a `with AncillaryFeature` context manager is something that we will definitely need e.g. for the `PluginFeature`.

Reply:
> Are you worried about cleanup after failed tests?

I knew that it executed in this way. My comment was more of a:
"I though pytest yield fixtures were a context manager and handled the clean up themselves." But now that I think about it, that would be a very far reaching and possible very complicated (perhaps impossible) thing to do.

Reply:
Ah I see now what you mean. Yes, a generally applicable "auto-teardown" would be very difficult.